metals/src/main/scala/scala/meta/internal/metals/Diagnostics.scala

```diff
@@ -214,8 +214,7 @@ final class Diagnostics(
         range,
         d.getMessage,
         d.getSeverity,
-        d.getSource,
-        d.getCode
```

Comment:
In the spec it says that code `can be omitted`, and I don't believe we use it at all. Plus, now it's a `Either[String, Number]`, but it was also returning `null` often, so I just removed it. Let me know if I'm missing something, and we actually need this.

Reply:
There is a comment in conversion `bsp.Diagnostics -> lsp.Diagnostics`:
```
        // We omit diag.getCode since Bloop's BSP implementation uses 'code' with different semantics
        // than LSP. See https://github.com/scalacenter/bloop/issues/1134 for details
```
Issue is still open but was assigned 4 days ago, maybe it is worth just doing `Left(d.getCode)` ? (more complex as it is java and must handle null, but concept the same)
This way when you update conversion bsp->lsp it will be pushed to client.

What is this code for is actually a good question anyway.

Reply:
Thanks for pointing out the conversation. I'm hesitant to change something to account for the changes in bloop when they aren't there yet. Since on the metals side we aren't using `code` for anything (again as far as I can tell), do you think it might be best to leave it out for now and first implement it if needed when there when it's finished and concrete just to ensure we don't have to potentially make the changes for the current case, and then change them again later? I'd rather just do that once.

Reply:
I think we can drop code for now, it's not really useful anyway. This usually just means the code for the error/warning type. That can be useful in some automatic tooling, but here we just need to present data to the user. Let's drop it.